core-web/apps/dotcms-ui/src/app/view/components/dot-content-compare/components/dot-content-compare-editor/dot-content-compare-editor.component.ts

```diff
@@ -0,0 +1,49 @@
+import { AfterViewInit, Component, ElementRef, Input, ViewChild } from '@angular/core';
+
+import { DotContentCompareTableData } from '@components/dot-content-compare/store/dot-content-compare.store';
+import { DotDiffPipe } from '@dotcms/app/view/pipes';
+import { DotBlockEditorComponent } from '@dotcms/block-editor';
+
+@Component({
+    selector: 'dot-content-compare-editor',
+    templateUrl: './dot-content-compare-editor.component.html'
+})
+export class DotContentCompareEditorComponent implements AfterViewInit {
+    @ViewChild('blockEditor') blockEditor: DotBlockEditorComponent;
+    @ViewChild('blockEditorCompare') blockEditorCompare: DotBlockEditorComponent;
+    @ViewChild('HtmlCompare') htmlCompare: ElementRef;
+    @ViewChild('HtmlWorking') htmlWorking: ElementRef;
+
+    @Input() data: DotContentCompareTableData;
+    @Input() showDiff: boolean;
+    @Input() field: string;
+    @Input() label: boolean;
+
+    private dotDiffPipe = new DotDiffPipe();
+
+    htmlCompareValue: string;
+    htmlWorkingValue: string;
+
+    ngAfterViewInit(): void {
+        this.blockEditor.value = this.data.working[this.field];
+        this.blockEditor?.valueChange.subscribe(() => {
+            this.htmlWorkingValue = this.blockEditor.editor.getHTML();
+
+            if (this.htmlWorking) {
+                this.htmlWorking.nativeElement.innerHTML = this.htmlWorkingValue;
+            }
+        });
+
+        this.blockEditorCompare.value = this.data.compare[this.field];
+        this.blockEditorCompare?.valueChange.subscribe(() => {
+            this.htmlCompareValue = this.blockEditorCompare.editor.getHTML();
+            if (this.htmlCompare) {
+                this.htmlCompare.nativeElement.innerHTML = this.dotDiffPipe.transform(
+                    this.htmlCompareValue,
+                    this.htmlWorkingValue,
+                    true
+                );
+            }
+        });
```

Comment:
I think we should get rid of the pipe inside of the component, maybe you can transform this subscribe in a async pipe doing something like:

```javascript
valueChanges$ = this.blockEditorCompare?.valueChange.pipe(switchMap(() => of(this.blockEditorCompare.editor.getHTML()))
```

And then in the HTML we can do something like:

```html
<ng-container *ngIf="valueChanges$ | async as compareValue">
    <div #HtmlWorking *ngIf="label" data-testId="table-editor-label"></div>
    <div #HtmlCompare *ngIf="!label" data-testId="table-editor-compare">
        {{compareValue | dotDiff: 'htmlWorkingValue':'true'}}
    </div>
</ng-container>

<dot-block-editor id="blockEditor" #blockEditor [ngStyle]="{ display: 'none' }"> </dot-block-editor>
<dot-block-editor id="blockEditorCompare" #blockEditorCompare [ngStyle]="{ display: 'none' }">
</dot-block-editor>
```

@fmontes what do you think about this change?

Not really sure if this code works, so you can test it and if you need any help ping me

Reply:
@manuelrojas @zJaaal If Angular removes the `iframe` it may be because it considered it as `dangerous HTML`, you can prevent that behavior by using sanitizer:

```ts
this.html = this.sanitizer.bypassSecurityTrustHtml(html);
```

SonarQube will mark it as `suspicious/smell` but that `HTML` is handled by us -`Block Editor`-. So, as soon as the user is not able to edit the `HTML` directly,  it should be fine.